imagery/i.svm.predict/main.c

```diff
@@ -0,0 +1,419 @@
+
+/****************************************************************************
+ *
+ * MODULE:       i.svm.predict
+ * AUTHOR(S):    Maris Nartiss - maris.gis gmail.com
+ * PURPOSE:      Predicts values with Support Vector Machine classifier
+ *
+ * COPYRIGHT:    (C) 2021 by Maris Nartiss and the GRASS Development Team
+ *
+ *               This program is free software under the GNU General Public
+ *               License (>=v2). Read the file COPYING that comes with GRASS
+ *               for details.
+ *
+ *               Development of this module was supported from
+ *               science funding of University of Latvia (2020/2021).
+ * 
+ *****************************************************************************/
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <unistd.h>
+
+#include <libsvm/svm.h>
+
+#include <grass/gis.h>
+#include <grass/raster.h>
+#include <grass/imagery.h>
+#include <grass/glocale.h>
+
+#define XSTR(s) STR(s)
+#define STR(s) #s
+
+
+/* LIBSVM message wrapper */
+void print_func(const char *s)
+{
+    G_verbose_message("%s", s);
+};
+
+
+int main(int argc, char *argv[])
+{
+    struct GModule *module;
+    struct Option *opt_group, *opt_subgroup, *opt_sigfile, *opt_values;
+    struct Option *opt_svm_cache_size;
+
+    char name_values[GNAME_MAX], name_sigfile[GNAME_MAX];
+    char name_group[GNAME_MAX], name_subgroup[GNAME_MAX];
+    char mapset_values[GMAPSET_MAX], mapset_sigfile[GMAPSET_MAX];
+    char mapset_group[GMAPSET_MAX], mapset_subgroup[GMAPSET_MAX];
+    char sigfile_dir[GPATH_MAX], model_file[GPATH_MAX];
+
+    struct Ref group_ref;
+    char **names_ordered, **mapsets_ordered;
+    const char **semantic_labels_group, **semantic_labels_svm;
+    char semantic_label[GNAME_MAX];
+    int semantic_label_count = 0, semantic_label_match_count = 0;
+
+    struct svm_model *model;
+
+    struct History history;
+    FILE *misc_file;
+    int sigfile_version;
+    double *rescale, r;
+    int rescale_count = 0;
+
+    G_gisinit(argv[0]);
+
+    module = G_define_module();
+    G_add_keyword(_("imagery"));
+    G_add_keyword(_("svm"));
+    G_add_keyword(_("classification"));
+    G_add_keyword(_("prediction"));
+    G_add_keyword(_("regression"));
+    module->description = _("Predict with a SVM");
```

Comment:
> I'm not sure if a...machine or machines is correct here.

Good question as technically the module provides several SVM algorithms. Lets keep singular as prediction is made with a single SVM algorithm although the module supports multiple.

Reply:
> libsvm might need to be also mentioned since it is part of the signature file identification, no?

Should it? IMHO we keep internals for documentation and just provide general terms for main UI parts.